test/uk/gov/hmrc/minorentityidentificationfrontend/services/RegistrationOrchestrationServiceSpec.scala

```diff
@@ -81,4 +81,50 @@ class RegistrationOrchestrationServiceSpec extends AnyWordSpec with Matchers wit
     }
   }
 
+  "register Unincorporated association" should {
+    "return Registered" when {
+      "the user has successfully passed BV check" in {
+        mockStorageService.retrieveBusinessVerificationStatus(testJourneyId) returns Future.successful(Some(BusinessVerificationPass))
+        mockRegistrationConnector.registerUA(testCtutr, testRegime) returns Future.successful(Registered(testSafeId))
+        mockStorageService.storeRegistrationStatus(testJourneyId, Registered(testSafeId)) returns Future.successful(SuccessfullyStored)
+
+        mockAuditService.auditJourney(testJourneyId, testUAJourneyConfig()) returns Future.successful(())
+
+        val result = await(TestRegistrationOrchestrationService.register(testJourneyId, Some(testCtutr), testUAJourneyConfig()))
+
+        result mustBe Registered(testSafeId)
+
+        mockAuditService.auditJourney(testJourneyId, testUAJourneyConfig()) was called
+      }
+      "the Business Verification Check is disabled" in {
+        mockStorageService.retrieveBusinessVerificationStatus(testJourneyId) returns Future.successful(None)
+        mockRegistrationConnector.registerUA(testCtutr, testRegime) returns Future.successful(Registered(testSafeId))
+        mockStorageService.storeRegistrationStatus(testJourneyId, Registered(testSafeId)) returns Future.successful(SuccessfullyStored)
+
+        mockAuditService.auditJourney(testJourneyId, testUAJourneyConfig(false)) returns Future.successful(())
```

Comment:
can we used named parameter .... reading it you cannot understand what false means